lychee-lib/src/types/cache.rs

```diff
@@ -1,13 +1,13 @@
 use std::fmt::Display;
 
-use serde::{Deserialize, Serialize};
+use serde::{Deserialize, Deserializer, Serialize};
 
 use crate::{ErrorKind, Status};
 
 /// Representation of the status of a cached request. This is kept simple on
 /// purpose because the type gets serialized to a cache file and might need to
 /// be parsed by other tools or edited by humans.
-#[derive(Debug, Serialize, Deserialize, Hash, PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
+#[derive(Debug, Serialize, Hash, PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
 pub enum CacheStatus {
```

Comment:
If using serde attributes like my comment you referenced showed:

```rust
#[serde(tag = "t", content = "c")]
pub enum CacheStatus {
```

This would include the enum type/tag, followed by the value (if any). Although I noted at the end of my comment that the default deserializer had trouble for some reason.

Reply:
Unfortunately I also could not get that to work with the default deserializer and I was also weary of making any "breaking changes" to the `.lycheecache` file format